crates/sui-indexer/src/models/tx_indices.rs

```diff
@@ -101,37 +114,68 @@ impl TxIndex {
                 cp_sequence_number,
                 tx_sequence_number,
                 recipient: s.to_vec(),
+                transaction_kind,
             })
             .collect();
-        let tx_input_objects = self
-            .input_objects
-            .iter()
-            .map(|o| StoredTxInputObject {
-                cp_sequence_number,
-                tx_sequence_number,
-                object_id: bcs::to_bytes(&o).unwrap(),
-            })
-            .collect();
-        let tx_changed_objects = self
-            .changed_objects
-            .iter()
-            .map(|o| StoredTxChangedObject {
-                cp_sequence_number,
-                tx_sequence_number,
-                object_id: bcs::to_bytes(&o).unwrap(),
-            })
-            .collect();
-        let tx_calls = self
-            .move_calls
-            .iter()
-            .map(|(p, m, f)| StoredTxCalls {
-                cp_sequence_number,
-                tx_sequence_number,
-                package: p.to_vec(),
-                module: m.to_string(),
-                func: f.to_string(),
-            })
-            .collect();
+
+        let mut address_rel_map: HashMap<&SuiAddress, i16> = HashMap::new();
+        for sender in &self.senders {
+            address_rel_map.entry(sender).or_insert(0);
+        }
+
+        for recipient in &self.recipients {
```

Comment:
strictly sender: rel = 0
strictly recipient: rel = 2
sender and recipient: rel = 1

This allows us to use a range filter `rel <= 1` for sender, `rel >= 1` for recipient which postgres seems to work better with (w.r.t leveraging indexes on address and rel.) Conversely, our initial approach of using or filters led postgres to do wild things like bitmap heap scans.